vellum/workflows/post_twitter/nodes/post_to_twitter.py

```diff
@@ -21,16 +21,50 @@ def run(self) -> Outputs:
         if not twitter_app:
             raise ValueError("Twitter application not found in database")
         
-        api_key = twitter_app.client_id
-        api_secret = twitter_app.client_secret
+        if not twitter_app.client_id or not twitter_app.client_secret:
+            raise ValueError("Twitter API consumer key and secret not configured")
+        
+        if not twitter_app.access_token or not twitter_app.refresh_token:
+            raise ValueError("Twitter access token and secret not configured")
         
         tweet_text = self.selected_tweet.text
         if self.selected_tweet.hashtags:
             hashtag_text = " " + " ".join(f"#{tag}" for tag in self.selected_tweet.hashtags)
             if len(tweet_text + hashtag_text) <= 280:
                 tweet_text += hashtag_text
         
-        tweet_id = "mock_tweet_id"
-        summary = f"Mock tweet prepared: '{tweet_text}'"
-        logger.info(summary)
-        return self.Outputs(tweet_id=tweet_id, summary=summary)
+        try:
+            client = tweepy.Client(
+                consumer_key=twitter_app.client_id,
+                consumer_secret=twitter_app.client_secret,
+                access_token=twitter_app.access_token,
+                access_token_secret=twitter_app.refresh_token
+            )
+            
+            response = client.create_tweet(text=tweet_text)
+            tweet_id = response.data['id']
+            summary = f"Successfully posted tweet: '{tweet_text}' (ID: {tweet_id})"
+            logger.info(summary)
+            
+            return self.Outputs(tweet_id=str(tweet_id), summary=summary)
+            
+        except tweepy.Unauthorized as e:
+            error_msg = f"Twitter authentication failed: {str(e)}"
+            logger.error(error_msg)
+            raise ValueError(error_msg)
```

Comment:
Turn all of these ValueError's into NodeException's, with an appropriate error code that isn't INTERNAL_ERROR